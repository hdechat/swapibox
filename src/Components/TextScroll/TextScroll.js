import React from 'react'
import PropTypes from 'prop-types'
import './TextScroll.css'

const TextScroll = ({ textCrawl }) => {
  return (
    <div>
      <div className="textScroll">
        <p>{textCrawl.text}</p>
        <p>{textCrawl.title}</p>
        <p>{textCrawl.released}</p>
      </div>
      <button onClick={() => document.querySelector('aside').style.display = 'none'}>Click</button>
    </div>
  )
}

TextScroll.propTypes = {
  textCrawl: PropTypes.object.isRequired
}

export default TextScroll
