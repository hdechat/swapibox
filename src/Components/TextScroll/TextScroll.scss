.textcrawl-container {
  background: url("../../assets/starry-night.png");
  height: 100%;
  overflow: hidden;
  width: 100%;
}

.star-wars {
  border: none;
  color: #feda4a;
  display: flex;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  height: 800px;
  justify-content: center;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  position: relative;
  text-align: justify;
  width: 100%;
}

.crawl {
  position: relative;
  transform-origin: 50% 10%;
  animation: crawl 70s linear;
}



@keyframes crawl {
  0% {
    top: 10em;
    transform: rotateX(40deg) translateZ(0);
  }
  85% { 
    top: -6000px;
    transform: rotateX(0deg) translateZ(-2500px);
  }
}
