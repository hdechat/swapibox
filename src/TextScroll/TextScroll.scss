.textScroll{
  position: absolute;
  left: 30%;
  width: 40%;
  animation-name: crawl;
  animation-duration: 30s;
}

@keyframes crawl {
    0%   { bottom:0%; transform:perspective(0px) rotateX(0deg)}
    100% {bottom:80%; transform:perspective(0px) rotateX(0deg)}
}

