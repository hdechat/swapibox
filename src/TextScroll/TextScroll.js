import React from 'react';
import PropTypes from 'prop-types';
import './TextScroll.css';

const TextScroll = ({ textCrawl }) => {
  return (
    <div className="textScroll">
      <p>{textCrawl.text}</p>
      <p>{textCrawl.title}</p>
      <p>{textCrawl.released}</p>
    </div>
  )
}

TextScroll.propTypes = {
  textCrawl: PropTypes.object
}

export default TextScroll;
