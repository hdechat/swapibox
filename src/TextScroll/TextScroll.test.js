import React from 'react'
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import TextScroll from './TextScroll.js';

describe('TextScroll', () => {
  let textScroll;
  let mockTextCrawl = {
    'title': 'text',
    'text': 'text',
    'release': 'text'
  }

  beforeEach(() => {
    const textScroll = shallow(<TextScroll textCrawl={mockTextCrawl} />)
  });

  it('matches snapshot', () => {
    expect(textScroll).toMatchSnapshot();
  });
  
})
