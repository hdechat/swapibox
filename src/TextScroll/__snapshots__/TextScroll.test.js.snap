// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextScroll matches snapshot 1`] = `undefined`;
